### DISCUSSION TITLE
Add the possibility to only auto-deploy specific branches from GitHub

### DISCUSSION THREAD

**Discussion by josias-r** [vercel/vercel#4419] [Ideas] [answered]
Currently, you can only disable GitHub auto deployments inside the `vercel.json`. It would be cool if you could also define a whitelist for branches that should get auto deployed (in my case I would only like to auto-deploy pushes to the master branch).

This allows you to make changes to branches during development phases w/out causing unnecessary builds.

Deployment hooks can solve my specific case, but it would be helpful if you could white- or even blacklist some branch names (preferably w/ glob pattern support).

This could be achieved by adding an optional array of strings to the github configurations inside `vercel.json`.

**Comment by paulogdm** [comment, marked as answer]
I completely agree. This is being worked on, and we are tracking it internally.

To suggest more features, use vercel.com/support.

**Reply by Mohamadhassan98** [reply]
Any updates on this feature so far?

**Reply by paulogdm** [reply]
Yep: https://vercel.com/knowledge/how-do-i-use-the-ignored-build-step-field-on-vercel

**Reply by leerob** [reply]
For visibility, this also exists now:

```
{
  "git": {
    "deploymentEnabled": {
      "main": false
    }
  }
}
```

> In case you want to turn auto-deployments off for every branch, property value must be set to false.

https://vercel.com/docs/projects/project-configuration/git-configuration

**Comment by GaddMaster** [comment]
Hi all
How am I supposed to understand this gobblygoop
https://vercel.com/knowledge/how-do-i-use-the-ignored-build-step-field-on-vercel
I just need build branch A, B and C, no previews.
How to do.
Thanks

**Reply by josias-r** [reply]
What's not to understand? There even is a how-to an example for your use-case:
```
#!/bin/bash

echo "VERCEL_GIT_COMMIT_REF: $VERCEL_GIT_COMMIT_REF"

if [[ "$VERCEL_GIT_COMMIT_REF" == "A" || "$VERCEL_GIT_COMMIT_REF" == "B" || "$VERCEL_GIT_COMMIT_REF" == "C"  ]] ; then
  # Proceed with the build
	echo "✅ - Build can proceed"
  exit 1;

else
  # Don't build
  echo "🛑 - Build cancelled"
  exit 0;
fi
```
Don't forget to:
> To run a bash script in the "Ignored Build Step", you need to set the following in the field: `bash script.sh`. Do notice the file should exist in your repository. An example of a bash script: ...



It might not be as simple as an array of branches, as my original question described, but that's just because the feature is more powerful than that ;)
